component/azstorage/block_blob.go

```diff
@@ -143,6 +145,25 @@ func (bb *BlockBlob) getCredential() azblob.Credential {
 	return cred.(azblob.Credential)
 }
 
+// NewPipeline creates a Pipeline using the specified credentials and options.
+func NewBlobPipeline(c azblob.Credential, o azblob.PipelineOptions, ro ste.XferRetryOptions) pipeline.Pipeline {
+	// Closest to API goes first; closest to the wire goes last
+	f := []pipeline.Factory{
+		azblob.NewTelemetryPolicyFactory(o.Telemetry),
+		azblob.NewUniqueRequestIDPolicyFactory(),
+		ste.NewBlobXferRetryPolicyFactory(ro),
+	}
+	f = append(f, c)
+	f = append(f,
+		pipeline.MethodFactoryMarker(), // indicates at what stage in the pipeline the method factory is invoked
+		ste.NewRequestLogPolicyFactory(ste.RequestLogOptions{
+			LogWarningIfTryOverThreshold: o.RequestLog.LogWarningIfTryOverThreshold,
+			SyslogDisabled:               o.RequestLog.SyslogDisabled,
+		}))
+
+	return pipeline.NewPipeline(f, pipeline.Options{HTTPSender: o.HTTPSender, Log: o.Log})
```

Comment:
As many of these init flows have changed, we shall check our sdk logging is working fine or not.
Also, moving to latest version of sdk solves the retry issue or we have taken any specific step in this change set?

Reply:
We are now importing the ste RetryPolicy

Reply:
I created a run that dumps logs in the pipeline and I see all the SDK logs as usual